hack/deploy-transport-controller.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Transport deployment script assumes it runs within a kubeflex context.
+
+export WDS_NAME="$1" ## first argument is WDS name
+export IMBS_NAME="$2" ## second argument is IMBS name
+export TRANSPORT_CONTROLLER_IMAGE="$3" ## third argument is transport controller image
+export HOST_IP=$(ifconfig | grep -w inet | awk '{ print $2 }' | grep -v 127.0.0.1 | head -1)
+export IMAGE_PULL_POLICY="${IMAGE_PULL_POLICY:=Always}"
+
+SRC_DIR="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)"
+cpname='$cpname' get_incluster_key='$get_incluster_key' secret_name='$secret_name' secret_namespace='$secret_namespace' key='$key' envsubst < "${SRC_DIR}/../deploy/transport-controller-deployment.yaml.template"  | kubectl apply -f -
```

Comment:
The `cpname='$cpname' get_incluster_key='$get_incluster_key' secret_name='$secret_name' secret_namespace='$secret_namespace' key='$key'` part is awkward and fragile. It requires maintainers to keep it consistent with the script in the ConfigMap. I would use two `kubectl apply` commands, one using `envsubst` and one not. Or...

Wouldn't it be simpler to inline the content of that transport-controller-deployment.yaml.template file, making this script say

```shell
kubectl apply -f - <<EOF
(template goes here, using full ordinary shell syntax -- including \$ )
EOF
```

Reply:
I pushed a fix. 
apparently `envsubst` supports the notion of specifying a list of env vars to replace and it leaves all the others. 
so I just added the 5 env vars that are set in this script.
let me know if it makes sense.

Reply:
That is good enough.